src/operator/src/statement/copy_database.rs

```diff
@@ -0,0 +1,250 @@
+                .copy_table_to(
+                    CopyTableRequest {
+                        catalog_name: req.catalog_name.clone(),
+                        schema_name: req.schema_name.clone(),
+                        table_name,
+                        location: table_file,
+                        with: req.with.clone(),
+                        connection: req.connection.clone(),
+                        pattern: None,
+                        direction: CopyDirection::Export,
+                        timestamp_range: req.time_range,
+                    },
+                    QueryContextBuilder::default().build(),
+                )
+                .await?;
+            exported_rows += exported;
+        }
+        Ok(Output::AffectedRows(exported_rows))
+    }
+
+    /// Imports data to database from a given location and returns total rows imported.
+    #[tracing::instrument(skip_all)]
+    pub(crate) async fn copy_database_from(
+        &self,
+        req: CopyDatabaseRequest,
+        ctx: QueryContextRef,
+    ) -> error::Result<Output> {
+        // location must end with /
+        ensure!(
+            req.location.ends_with('/'),
+            InvalidCopyParameterSnafu {
+                key: "location",
+                value: req.location,
+            }
+        );
+
+        info!(
+            "Copy database {}.{} from dir: {}, time: {:?}",
+            req.catalog_name, req.schema_name, req.location, req.time_range
+        );
+        let suffix = Format::try_from(&req.with)
+            .context(error::ParseFileFormatSnafu)?
+            .suffix();
+
+        let entries = list_files_to_copy(&req, suffix).await?;
+
+        let continue_on_error = req
+            .with
+            .get(CONTINUE_ON_ERROR_KEY)
+            .and_then(|v| bool::from_str(v).ok())
+            .unwrap_or(false);
+        let mut rows_inserted = 0;
+
+        for e in entries {
+            let table_name = match parse_file_name_to_copy(&e) {
+                Ok(table_name) => table_name,
+                Err(err) => {
+                    if continue_on_error {
+                        error!(err; "Failed to import table from file: {:?}", e);
+                        continue;
+                    } else {
+                        return Err(err);
+                    }
+                }
+            };
+            let req = CopyTableRequest {
+                catalog_name: req.catalog_name.clone(),
+                schema_name: req.schema_name.clone(),
+                table_name: table_name.clone(),
+                location: format!("{}/{}", req.location, e.path()),
+                with: req.with.clone(),
+                connection: req.connection.clone(),
+                pattern: None,
+                direction: CopyDirection::Import,
+                timestamp_range: None,
+            };
+            debug!("Copy table, arg: {:?}", req);
+            match self.copy_table_from(req, ctx.clone()).await {
```

Comment:
```suggestion
            match self.copy_table_from(req, ctx).await {
```

Reply:
> 

ctx is used in a for-loop, so a clone is required.